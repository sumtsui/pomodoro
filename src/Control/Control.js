import React from 'react';
import Proptypes from 'prop-types';
import {Link} from 'react-router-dom';

const Control = ({running, pauseCountDown, endCountDown, paused, startCountDown, onFinish}) => {
  return (
    running ?
      <div className='button-running'>
        <button onClick={pauseCountDown}>{paused ? 'Continue' : 'Pause'}</button>
        <button onClick={endCountDown}>Skip</button>
        <Link to='/finish'>
          <button onClick={onFinish}>
            End
          </button>
        </Link>
      </div >
      :
      <div>
        <button
          className='start-button'
          onClick={startCountDown} >
          Start
        </button>
      </div>
  )
}

Control.prototype = {
  running: Proptypes.string.isRequired,
  paused: Proptypes.string.isRequired,
  pauseCountDown: Proptypes.func.isRequired,
  endCountDown: Proptypes.func.isRequired,
  startCountDown: Proptypes.func.isRequired
}

export default Control;