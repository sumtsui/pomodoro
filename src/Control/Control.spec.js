import React from 'react';
import {shallow, mount} from 'enzyme';
import Control from './Control';

describe('Render', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Control/>)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  })

  it('should render "Pause" and "End" buttons if the counter is running', () => {
    wrapper.setProps({running: true});
    expect(wrapper.find('.button-running').text()).toEqual('PauseEnd')
  })

  it('should render "Start" buttons if the counter is running', () => {
    wrapper.setProps({running: false});
    expect(wrapper.find('button').text()).toEqual('Start')
  })

  test('when counter is paused, Pause button should turn to "Continue"', () => {
    wrapper.setProps({
      running: true,
      paused: true
    });
    expect(wrapper.find('.button-running').text()).toEqual('ContinueEnd');
  })
})

describe('Start button', () => {
  test('onClick function is called when the button is clicked', () => {
    const fn = jest.fn();
    const component = shallow(
      <Control
        startCountDown={fn}
      />
    )
    component.find('.start-button').simulate('click');
    expect(fn.mock.calls.length).toBe(1);
  })

})
