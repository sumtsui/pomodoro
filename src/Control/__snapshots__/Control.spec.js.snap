// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="start-button"
        onClick={undefined}
>
        Start
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Start",
        "className": "start-button",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "Start",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="start-button"
          onClick={undefined}
>
          Start
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Start",
          "className": "start-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Start",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
